src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -488,8 +490,9 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
         super.transform(treeCopy.ArrayValue(rest, rest.elemtpt, elem0 :: rest.elems))
 
       // List(a, b, c) ~> new ::(a, new ::(b, new ::(c, Nil)))
+      // Seq(a, b, c) ~> new ::(a, new ::(b, new ::(c, Nil)))
       case Apply(appMeth, List(Apply(wrapArrayMeth, List(StripCast(rest @ ArrayValue(elemtpt, _))))))
-      if wrapArrayMeth.symbol == currentRun.runDefinitions.Predef_wrapRefArray && appMeth.symbol == List_apply && rest.elems.length < transformListApplyLimit =>
+        if wrapArrayMeth.symbol == currentRun.runDefinitions.Predef_wrapRefArray && (appMeth.symbol == listApply || appMeth == seqApply) && rest.elems.length < transformListApplyLimit =>
```

Comment:
```suggestion
        if wrapArrayMeth.symbol == currentRun.runDefinitions.Predef_wrapRefArray && (appMeth.symbol == listApply || appMeth.symbol == seqApply) && rest.elems.length < transformListApplyLimit =>
```

So should probably include the allocation test that's in #8789